packages/protocol/contracts/layer1/based/TaikoInbox.sol

```diff
@@ -176,6 +176,7 @@ abstract contract TaikoInbox is EssentialContract, ITaikoInbox, IProposeBatch, I
                 blobByteOffset: params.blobParams.byteOffset,
                 blobByteSize: params.blobParams.byteSize,
                 gasLimit: config.blockMaxGasLimit,
+                baseFee: params.baseFee,
```

Comment:
Questions for @smartprogrammer93

1. Is there a contract or mechanism that deterministically calculates params.baseFee, so the proposer can’t arbitrarily set it too high?
1. If the baseFee parameter is applied immediately to the current batch, users won’t be able to predict it in advance -- meaning they won’t know what base fee to use for their transactions to be included in this block. Is that correct?

Reply:
With the current base fee design, users specify a max base fee. If the proposer selects transactions where the lowest max base fee is X, they can simply set the base fee to X to maximize revenue. Does this behavior break the user’s assumption that they might pay less than their max?